src/main/java/cn/nukkit/entity/EntityCreature.java

```diff
@@ -31,6 +31,11 @@ protected boolean applyNameTag(Player player, Item item){
         if (item.hasCustomName()) {
             this.setNameTag(item.getCustomName());
             this.setNameTagVisible(true);
+            int g = player.getGamemode();
+
+            if(g == Player.ADVENTURE || g == Player.SURVIVAL) {
+                player.getInventory().removeItem(item);
+            }
             player.getInventory().removeItem(item);
```

Comment:
Duplicate

Reply:
wdym?

I've chosen to move the behaviour into EntityCreature rather than ItemNameTag as the Player parameter is available.
Removed it from EntityMob + EntityAnimal

Reply:
You call `removeItem` twice

Reply:
oh I'm blind lmao